app/widget/slider/sliderbase.cpp

```diff
@@ -401,4 +450,11 @@ void SliderBase::RepositionLadder()
   }
 }
 
+void SliderBase::ChangeSliderType()
+{
+  if (mode_ == kRational) {
+    emit changeRationalDisplayType();
```

Comment:
Would prefer the RationalSlider do this itself than have to go through SliderBase, but you might need to add a `protected` accessor so that RationalSlider can get it (something like `QLabel *label() { return label_; }`)